### ISSUE TITLE
[BUG] Applying help_option_names to the Typer instance is inconsistent

### ISSUE THREAD

**Issue by richieadler** [tiangolo/typer#208]
### Description of the bug

The results of applying `help_option_names` to the **Typer** instance are inconsistent, depending whether one has only one command or more than one.

### To Reproduce

* Create a file `main.py` with:

```Python
import typer

CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])

app = typer.Typer(context_settings=CONTEXT_SETTINGS, add_completion=False)

@app.command()
def hello(nom: str):
    print(f"Hello {nom}")

@app.command()
def bye(nom: str):
    print(f"Bye {nom}")

if __name__ == '__main__':
    app()

```

* Call it with:

```bash
python main.py -h
```

* It outputs:

```
Usage: test_typer.py [OPTIONS] COMMAND [ARGS]...

Options:
  -h, --help  Show this message and exit.

Commands:
  bye
  hello
```

* However, if the whole function `bye` is commented out, the same execution results in:

```
Usage: test_typer.py [OPTIONS] NOM
Try 'test_typer.py --help' for help.

Error: no such option: -h
```

### Expected behavior

The message should be:

```
Usage: test_typer.py [OPTIONS] NOM

Arguments:
  NOM  [required]

Options:
  -h, --help  Show this message and exit.
```

### Environment

* OS: Windows 10 [Version 10.0.19041.685]
* Typer Version: 0.3.2
* Python version: Python 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)] on win32

**Comment by malthunayan** [comment]
@richieadler just add your context settings to the command instead of the `Typer`  instance if you have a single command and it should work as expected. In your example that would be:
```
import typer

CONTEXT_SETTINGS = dict(help_option_names=["-h", "--help"])

app = typer.Typer(add_completion=False)


@app.command(context_settings=CONTEXT_SETTINGS)
def hello(nom: str):
    print(f"Hello {nom}")


if __name__ == "__main__":
    app()
```

If you have multiple commands or a command and a callback, then just move your `CONTEXT_SETTINGS` back again to where you instantiate the `Typer` object. I hope this helps you out.

The reason why it works this way is because internally when `typer` creates the command above it creates a `click` command. If it is multiple commands then it creates a `click` group. When `typer` is creating the `click` group it is passing the entire `app` object in, however if it is a single command, then it just passes the registered command (not the entire `typer` instance).  This means you have to pass in your desired settings to the command object instead of the `typer` instance. I hope I made sense, if not, I'd be happy to answer more of your questions.

**Comment by daddycocoaman** [comment]
@malthunayan explained it very well. IMO though, the fact that the context settings DO get passed when multiple commands exists should mean they should still get passed when there's only one command, especially since creating the app object with a single command is a completely valid design. It's possible to do with just a few extra lines of code in how the single command is handled. 

I've already tested a change and I'm gonna put in a PR later today but not sure if it'll be implemented anytime soon. :P But in the meantime, adding the context_settings to the command, not the typer app, is the way go.

**Comment by tiangolo** [comment]
Thanks for the detailed description and example! :bow: 

This was fixed by @daddycocoaman here: https://github.com/tiangolo/typer/pull/210

It will be available in Typer version `0.4.2` released in some hours. :tada:
